### PR TITLE
[MUI] Migration Guide for Unified Theme & Material UI v5

### PR DESCRIPTION
**Pull request by tudi2d**
New version of https://github.com/backstage/backstage/pull/18098 due to `force` push.

**What**: Creating a seperate, short documentation for migrating to MUI v5 (https://github.com/backstage/backstage/pull/17696#discussion_r1212114911)

**Why**: Moving this to a seperate PR to ship it faster, align with potential issues raised on the migration & additionally gather feedback if this is the right approacht for documentation or if something else is needed.

Follow up: 
- Update GraphiQL plugin PR
- Update https://backstage.io/docs/getting-started/app-custom-theme to use `createUnifiedTheme`

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by tudi2d** [comment]
Added @awanlin & @Rugvip as they reviewed previous attempt.

**Review by awanlin — APPROVED** [review]
Thanks again for putting this together @tudi2d! 🚀 

Once thing I noticed is that we don't have this in the sidebar? Is your intention to just get this in place and then iterate on it before adding it to the sidebar?

**Comment by tudi2d** [comment]
Ah no - will add it to the sidebar. Thanks for the heads up! I want to get this in & then create a follow up PR to adjust the theme customisation documentation.

File: mkdocs.yml [hunk 1]
```diff
@@ -193,6 +193,7 @@ nav:
       - Switching Backstage from SQLite to PostgreSQL: 'tutorials/switching-sqlite-postgres.md'
       - Using the Backstage Proxy from Within a Plugin: 'tutorials/using-backstage-proxy-within-plugin.md'
       - Migration to Yarn 3: 'tutorials/yarn-migration.md'
+      - Migration to Material UI v5: 'tutorials/migrate-to-mui5.md'
```
Comment (awanlin) [inline mkdocs.yml:196]: This is great but you'll also want to update this file too: https://github.com/backstage/backstage/blob/00946caf9ad08b3103827779b7375763a32a3278/microsite/sidebars.json#L402

**Review by awanlin — APPROVED** [review]

**Review by camilaibs — APPROVED** [review]
Thanks for putting it together 🎉 
Left minor comments:

File: docs/tutorials/migrate-to-mui5.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+const app = createApp({
+  // ...
+  themes: [
+    {
+      // ...
+      provider: ({ children }) => (
+-       <ThemeProvider theme={lightTheme}>.
+-         <CssBaseline>{children}</CssBaseline>.
+-       </ThemeProvider
++       <UnifiedThemeProvider theme={builtinThemes.light} children={children} />
```
[the remark this answers is not in the document]
Comment (tudi2d) [reply docs/tutorials/migrate-to-mui5.md:21]: Yes. - might be helpfull to add an `import` statement to make clear where it comes from.

File: docs/tutorials/migrate-to-mui5.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+To comply with Material UI recommendations, we are enforcing a new linting rule that favors standard imports over named imports and also restricts 3rd-level imports as they are considered private ([Guide: Minimizing Bundle Size](https://mui.com/material-ui/guides/minimizing-bundle-size)).
+
+Important to keep in mind is that Material UI v5 is meant to be used with React Version 17 or higher. This means if you intend to use the Material UI v5 components in your plugins, you have to enforce React Version to be at least 17 for these plugins as well.
+
+There are `core-components` as well as components exported from `*-react` plugins written in Material UI v4, which expect Material UI components as props. In these cases you will still be forced to import Material UI v4 components.
+
+For current known issues with the Material UI v5 migration, follow our [Milestone on GitHub](https://github.com/backstage/backstage/milestone/40). Please open a new issue if you run into different problems.
+
+### Plugins
```
[the remark this answers is not in the document]
Comment (camilaibs) [reply docs/tutorials/migrate-to-mui5.md:40]: Yeah, for plugins, great to hear codemod is taking care of it, so all good then 🙌🏻

File: docs/tutorials/migrate-to-mui5.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+It is worth noting that we are still using `@mui/styles` & `jss`. You may stumble upon documentation for migrating to `emotion` when using `makeStyles` or `withStyles`. It is not necessary to switch to `emotion`.
+
+To comply with Material UI recommendations, we are enforcing a new linting rule that favors standard imports over named imports and also restricts 3rd-level imports as they are considered private ([Guide: Minimizing Bundle Size](https://mui.com/material-ui/guides/minimizing-bundle-size)).
+
+Important to keep in mind is that Material UI v5 is meant to be used with React Version 17 or higher. This means if you intend to use the Material UI v5 components in your plugins, you have to enforce React Version to be at least 17 for these plugins as well.
+
+There are `core-components` as well as components exported from `*-react` plugins written in Material UI v4, which expect Material UI components as props. In these cases you will still be forced to import Material UI v4 components.
```
[the remark this answers is not in the document]
Comment (tudi2d) [reply docs/tutorials/migrate-to-mui5.md:36]: The migration guide is just a few lines bellow - I think it is ok :)

**Review by benjdlambert — APPROVED** [review]

**Comment by awanlin** [comment]
Nice very happy to see this merged, many thanks @tudi2d 🚀

**Pull request merged**